File: SolverSeged/SolverSeged/Sigmoid.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SolverSeged
{
    class Sigmoid : ActivationStrategy
    {
        public override double Activation(double input)
        {
            return (1.0 / (1.0 + Math.Pow(Math.E, -input)));
        }

        public override double DeActivation(double output)
        {
            return (output * (1.0 - output));
        }

        public override string ToString()
        {
            return "Sigmoid";
        }
    }
}

File: ProblemClassifier/src/Classes/Sigmoid.java
package Classes;

/**
 * @author Danisovszky Márk
 */
public class Sigmoid extends ActivationFunction {

    @Override
    public double Activation(double input) {
        return (1.0 / (1.0 + Math.exp(-input)));
    }

    @Override
    public double DActivation(double input) {
        return (input * (1.0 - input));
    } 
}
